functorch/_src/memory_efficient_op_authoring.py

```diff
@@ -0,0 +1,159 @@
+import torch
+from torch import fx
+from torch.fx.proxy import GraphAppendingTracer
+from typing import Iterable
+from .eager_compilation import compiled_function, partition_with_recompute_fwd_in_bwd
+
+
+def tanh_backward_decomposition(out_grad, y):
+    return torch.sub(out_grad, out_grad * y * y)
+
+
+def sigmoid_backward_decomposition(out_grad, y):
+    # Computation - out_grad * (y * (1 - y))
+    return out_grad * (y - y * y)
+
+
+def softplus_backward_decomposition(out_grad, x, beta, threshold, out):
+    return out_grad * torch.sigmoid(x)
+
+
+def noop(x):
+    return x
+
+
+def _s_where_decomposition(a, b, c):
+    return torch.where(a, b, c)
+
+
+decomposition_rules = {}
+decomposition_rules[torch.ops.aten.tanh_backward] = tanh_backward_decomposition
+decomposition_rules[torch.ops.aten._s_where] = _s_where_decomposition
+decomposition_rules[torch.ops.aten.sigmoid_backward] = sigmoid_backward_decomposition
+decomposition_rules[torch.ops.aten.softplus_backward] = softplus_backward_decomposition
+decomposition_rules[torch.ops.aten.detach] = noop
+
+# TODO - Move to the python key based decomposition
+def decompose(fx_module):
```

Comment:
I wonder whether we should switch over to a "proper" decomposition implementation first.

Reply:
Eh... we can just do that later.